### ISSUE TITLE
Upgrade may fail when running the higstate if apiserver is not ready

### ISSUE THREAD

**Issue by charlesprost** [scality/metalk8s#1839]
Probably relates to #1836 

**Component**:

'containers', 'upgrade', 'ci', 'tests'

**What happened**:

When running the highstate during the upgrade test, we face this error:
```
                          salt_|-Run the highstate_|-Run the highstate_|-state:
                              ----------
                              __id__:
                                  Run the highstate
                              __jid__:
                                  20190927150938505970
                              __run_num__:
                                  3
                              __sls__:
                                  metalk8s.orchestrate.deploy_node
                              changes:
                                  ----------
                                  out:
                                      highstate
                                  ret:
                                      ----------
                                      bootstrap:
                                          - Rendering SLS 'metalk8s-2.2.0-dev:metalk8s.salt.minion.configured' failed: Jinja variable 'dict object' has no attribute 'ip'
                              comment:
                                  Run failed on minions: bootstrap
                              duration:
                                  8212.407
                              name:
                                  Run the highstate
                              result:
                                  False
                              start_time:
                                  15:09:38.391750
```

See https://eve.devsca.com/github/scality/metalk8s/#/builders/25/builds/207/steps/17/logs/stdio

See sosreport here: [sosreport-bootstrap-2019-09-27-rznleio.zip](https://github.com/scality/metalk8s/files/3689926/sosreport-bootstrap-2019-09-27-rznleio.zip)

When the Salt states are rendered, the Salt master uses the external pillar to retrieve the IP address, and for this it calls the apiserver to find the endpoint. But if the apiserver is not available, then it fails and the pillar contains no IP.

See the start time of the salt step at `15:09:38.391750`
And we only wait for APIserver to be available at `15:09:46.605898`

May be fixed by #1836 

We have a similar one here https://eve.devsca.com/github/scality/metalk8s/#/builders/25/builds/375/steps/17/logs/stdio:
```
                                          - Rendering SLS 'metalk8s-2.3.0-dev:metalk8s.salt.minion.configured' failed: Jinja variable 'dict object' has no attribute 'ip'
                                          - Rendering SLS 'metalk8s-2.3.0-dev:metalk8s.repo.offline' failed: Jinja variable 'dict object' has no attribute 'ports'
                                          - Rendering SLS 'metalk8s-2.3.0-dev:metalk8s.container-engine.containerd.installed' failed: Jinja variable 'dict object' has no attribute 'ports'
```

**What was expected**:

The apiserver should have been available, and the pillar value should be OK.

**Steps to reproduce**

Not sure since it's flaky.

**Resolution proposal** (optional):*

Wait for apiserver to be available before rendering the states.

**Comment by TeddyAndrieux** [comment]
`refresh_pillar` is not enough

**Comment by charlesprost** [comment]
@TeddyAndrieux we had a flaky with the same error message yesterday: `Jinja variable 'dict object' has no attribute 'ip'`: https://eve.devsca.com/github/scality/metalk8s/#/builders/1/builds/13596

**Comment by TeddyAndrieux** [comment]
Weird this one should have been fixed by change of this ticket + #1918
